tools/integration_tests/list_large_dir/list_dir_with_twelve_thousand_files_test.go

```diff
@@ -0,0 +1,141 @@
+// Copyright 2023 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package list_large_dir_test
+
+import (
+	"os"
+	"path"
+	"sort"
+	"strconv"
+	"testing"
+
+	"github.com/googlecloudplatform/gcsfuse/tools/integration_tests/util/setup"
+)
+
+func differentiateFileAndDirName(objs []os.DirEntry) (dirName []string, fileName []string) {
+	for _, obj := range objs {
+		if obj.IsDir() {
+			dirName = append(dirName, obj.Name())
+		} else {
+			fileName = append(fileName, obj.Name())
+		}
+	}
+	return
+}
+
+func findIfCorrectObjectExistInList(objs []string, obj string, t *testing.T) {
+	index := sort.SearchStrings(objs, obj)
+	if objs[index] != obj {
+		t.Errorf("Correct object does not exist.")
+	}
+}
+
+// Test with a bucket with twelve thousand files.
+func TestDirectoryWithTwelveThousandFiles(t *testing.T) {
+	dirPath := path.Join(setup.MntDir(), DirectoryWithTwelveThousandFiles)
+
+	objs, err := os.ReadDir(dirPath)
+	if err != nil {
+		t.Errorf("Error in listing directory.")
+	}
+
+	// number of objs - 12000
+	if len(objs) != NumberOfFilesInDirectoryWithTwelveThousandFiles {
+		t.Errorf("Listed incorrect number of files from directory: %v, expected 12000", len(objs))
+	}
+
+	// Checking if all the object is File type.
+	for i := 0; i < len(objs); i++ {
+		if objs[i].IsDir() {
+			t.Errorf("Listes object is incorrect.")
+		}
+	}
+
+	// Checking if correct file present in bucket.
+	var fileName []string
+	for i := 0; i < len(objs); i++ {
```

Comment:
nit: 

IIUC
(a) This code runs in 0(nlogn) ?
(b) It doesn't check duplicate file names which should be fine because there can't be duplicate files in filesystem ?
(c) File name is of type <filenameprefix>{i} where i >=1 & i <= 12000 ?

if above is correct, then can we reduce the complexity to O(n) ?

Reply:
Fixed